src/main/scala/com/etsy/sbt/Checkstyle.scala

```diff
@@ -10,56 +10,110 @@ import sbt._
     val checkstyleConfig = SettingKey[File]("checkstyle-config", "The location of the checkstyle configuration file")
     val xsltTransformations = SettingKey[Option[Set[XSLTSettings]]]("xslt-transformations", "An optional set of XSLT transformations to be applied to the checkstyle output")
+    val checkstyleCheckSeverityLevel = SettingKey[String]("checkstyle-check-level", "Sets the severity levels which should fail the build")
   }
 
   /**
     * Runs checkstyle
-    * 
+    *
     * @param conf The configuration (Compile or Test) in which context to execute the checkstyle command
     */
   def checkstyleTask(conf: Configuration): Initialize[Task[Unit]] = Def.task {
+    val configFile = (checkstyleConfig in conf).value.getAbsolutePath
     val outputFile = (checkstyleTarget in conf).value.getAbsolutePath
+    val source = (javaSource in conf).value.getAbsolutePath
+
+    checkstyleMain(configFile, outputFile, source, xsltTransformations.value)
+  }
+
+  /**
+    * Runs checkstyle and fails the task if issues are found.
+    *
+    * Use setting 'checkstyleCheckSeverityLevel' to define what type of issues should break the build.
+    *
+    * @param conf The configuration (Compile or Test) in which context to execute the checkstyle command
+    */
+  def checkstyleCheckTask(conf: Configuration): Initialize[Task[Unit]] = Def.task {
+    val configFile = (checkstyleConfig in conf).value.getAbsolutePath
+    val outputFile = (checkstyleTarget in conf).value.getAbsolutePath
+    val source = (javaSource in conf).value.getAbsolutePath
+
+    checkstyleMain(configFile, outputFile, source, xsltTransformations.value)
+
+    if (file(outputFile).exists) {
+      val log = streams.value.log
+      val report = scala.xml.XML.loadFile(file(outputFile))
+      val checkstyleSeverityLevels = Seq("ignore", "info", "warning", "error")
+      val appliedCheckstyleSeverityLevels = checkstyleSeverityLevels.indexOf(checkstyleCheckSeverityLevel.value) match {
+        case i if i == -1 => Seq("error") // unknown level supplied, default to just error
+        case i => checkstyleSeverityLevels.drop(i)
+      }
+      var issuesFound = 0
+      (report \ "file").foreach { file =>
+        (file \ "error").foreach { error =>
+          val severity: String = error.attribute("severity").get.head.text
+          if (appliedCheckstyleSeverityLevels.contains(severity)) {
+            val lineNumber: String = error.attribute("line").get.head.text
+            val filename: String = file.attribute("name").get.head.text
+            val errorMessage: String = error.attribute("message").get.head.text
+            log.error("Checkstyle " + severity + " found in " + filename + ":" + lineNumber + ": " + errorMessage)
+            issuesFound += 1
+          }
+        }
+      }
+
+      if (issuesFound > 0) {
+        log.error(issuesFound + " issue(s) found in Checkstyle report: " + outputFile + "")
+        sys.exit(1)
```

Comment:
Ah, this looks like a difference between the interactive and non-interactive modes.  When run interactively SBT will suppress the stacktrace from `sys.error`, but otherwise will not.  Unfortunately there doesn't seem to be an option for always suppressing the stack trace for a given task.

I don't mind the stack trace, but I can see how that would clutter the output.  If not having the stack trace would be better for you I'm fine with leaving this as-is.